src/rating/Rating.js

```diff
@@ -138,25 +153,57 @@ export default class Rating extends Component {
 
   getCurrentRating() {
     const { value } = this.state;
-    const { imageSize, ratingCount } = this.props;
+    const { fractions, imageSize, ratingCount } = this.props;
     const startingValue = ratingCount / 2;
     let currentRating = 0;
 
     if (value > ratingCount * imageSize / 2) {
       currentRating = ratingCount;
     } else if (value > imageSize) {
-      currentRating = Math.ceil(startingValue + value / imageSize);
+      currentRating = startingValue + value / imageSize;
+      currentRating = !fractions
+        ? Math.ceil(currentRating)
+        : +currentRating.toFixed(fractions);
     } else if (value < -ratingCount * imageSize / 2) {
       currentRating = 0;
     } else if (value < imageSize) {
-      currentRating = Math.ceil(startingValue + value / imageSize);
+      currentRating = startingValue + value / imageSize;
+      currentRating = !fractions
+        ? Math.ceil(currentRating)
+        : +currentRating.toFixed(fractions);
     } else {
-      currentRating = Math.ceil(startingValue);
+      currentRating = !fractions
+        ? Math.ceil(startingValue)
+        : +startingValue.toFixed(fractions);
     }
 
     return currentRating;
   }
 
+  setCurrentRating(rating) {
+    const { imageSize, ratingCount } = this.props;
+    // `initialRating` corresponds to `startingValue` in the getter. Naming it
+    // differently here avoids confusion with `value` below.
+    const initialRating = ratingCount / 2;
+    let value = -ratingCount * imageSize / 2;
+
+    if (rating > ratingCount) {
+      value = ratingCount * imageSize / 2;
+    } else if (rating > ratingCount / 2) {
+      value = (rating - initialRating) * imageSize;
+    } else if (rating < 0) {
+      value = -ratingCount * imageSize / 2;
+    } else if (rating < ratingCount / 2) {
+      value = (rating - initialRating) * imageSize;
+    } else {
+      value = 0;
+    }
+
+    const { position } = this.state;
+    position.setValue({ x: value, y: 0 });
```

Comment:
why do we need to reset the value of `rating` when we set a rating? 

Once the rating is rendered, I would think that the UI for the user interaction should begin from that point and not 0. 

Please correct me if this is not the resulting implementation.